InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.RemoveById.cs

```diff
@@ -0,0 +1,181 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using InternTrack.Core.Api.Models.Interns;
+using InternTrack.Core.Api.Models.Interns.Exceptions;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using Xunit;
+
+namespace InternTrack.Core.Api.Tests.Unit.Services.Foundations.Interns
+{
+    public partial class InternServiceTests
+    {
+
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnRemoveByIdIfSqlExceptionOccursAndLogItAsync()
+        {
+            // given
+            Guid someInternId = Guid.NewGuid();
+            SqlException sqlException = GetSqlException();
+
+            var failedInternStorageException =
+                new FailedInternStorageException(sqlException);
+
+            var expectedInterndependencyException =
+                new InternDependencyException(failedInternStorageException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectInternByIdAsync(It.IsAny<Guid>()))
+                    .ThrowsAsync(sqlException);
+
+            //when
+            ValueTask<Intern> removeInternTask =
+                this.internService.RemoveInternByIdAsync(someInternId);
+
+            //then
+            await Assert.ThrowsAsync<InternDependencyException>(() =>
+                removeInternTask.AsTask());
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectInternByIdAsync(It.IsAny<Guid>()),
+                    Times.Once());
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogCritical(It.Is(SameExceptionsAs(
+                    expectedInterndependencyException))),
+                        Times.Once());
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowdependencyExceptionOnRemoveByIdIfDatabaseUpdateExceptionOccursAndLogItAsync()
+        {
+            // given
+            Guid someInternId = Guid.NewGuid();
+            var databaseUpdateException = new DbUpdateException();
+
+            var failedInternStorageException =
+                new FailedInternStorageException(databaseUpdateException);
+
+            var expectedInternDependencyException =
+                new InternDependencyException(failedInternStorageException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectInternByIdAsync(It.IsAny<Guid>()))
+                    .ThrowsAsync(databaseUpdateException);
+
+            // when 
+            ValueTask<Intern> removeInternTask =
+                this.internService.RemoveInternByIdAsync(someInternId);
+
+            //then
+            await Assert.ThrowsAsync<InternDependencyException>(() =>
+                removeInternTask.AsTask());
```

Comment:
Old version here. Move to new

Reply:
@ADILATIC this is the old version of handling exceptions like @ElbekDeveloper said - move to the new pattern - check other similar code for examples.